.Lib9c.Tests/Action/DPoS/Control/SlashCtrlTestBase.cs

```diff
@@ -0,0 +1,173 @@
+            => Sum(values, Asset.ConsensusToken);
+
+        protected static FungibleAssetValue SumNCG(IEnumerable<FungibleAssetValue> values)
+            => Sum(values, Asset.GovernanceToken);
+
+        protected static FungibleAssetValue GetPower(IWorldState worldState, Address validatorAddress)
+        {
+            return worldState.GetBalance(
+                address: validatorAddress,
+                currency: Asset.ConsensusToken);
+        }
+
+        protected static FungibleAssetValue GetNCG(IWorldState worldState, Address address)
+        {
+            return worldState.GetBalance(
+                address: address,
+                currency: Asset.GovernanceToken);
+        }
+
+        protected static FungibleAssetValue GetShare(IWorldState worldState, Address validatorAddress)
+        {
+            var validator = ValidatorCtrl.GetValidator(worldState, validatorAddress)!;
+            return validator.DelegatorShares;
+        }
+
+        protected static FungibleAssetValue GetSlashAmount(
+            FungibleAssetValue value,
+            BigInteger factor,
+            params FungibleAssetValue[] assetsToSlashFirst)
+        {
+            var total = assetsToSlashFirst.Aggregate(value, Sum);
+            var (amount, remainder) = total.DivRem(factor);
+            for (var i = 0; i < assetsToSlashFirst.Length; i++)
+            {
+                var (a, r) = assetsToSlashFirst[i].DivRem(factor);
+                amount -= a;
+            }
+
+            return amount;
+        }
+
+        protected static FungibleAssetValue[] GetSlashAmounts(FungibleAssetValue[] values, BigInteger factor)
+            => values.Select(value => GetSlashAmount(value, factor)).ToArray();
+
+        protected static FungibleAssetValue NextNCG()
+        {
+            return new FungibleAssetValue(
+                currency: Asset.GovernanceToken,
+                majorUnit: Random.Shared.Next(1, 100),
+                minorUnit: Random.Shared.Next(0, 100));
+        }
+
+        protected static FungibleAssetValue[] NextNCGMany(int length)
+        {
+            return Enumerable.Range(0, length)
+                .Select(_ => NextNCG())
+                .ToArray();
+        }
+
+        protected static FungibleAssetValue PowerFromNCG(FungibleAssetValue value)
+            => Asset.ConsensusFromGovernance(value);
+
+        protected static FungibleAssetValue[] PowerFromNCG(FungibleAssetValue[] values)
+            => values.Select(PowerFromNCG).ToArray();
+
+        protected static FungibleAssetValue ShareFromPower(FungibleAssetValue value)
+        {
+            if (!value.Currency.Equals(Asset.ConsensusToken))
+            {
+                throw new ArgumentException(
+                        message: $"'{value}' is not {nameof(Asset.ConsensusToken)}",
+                        paramName: nameof(value));
+            }
+
+            return FungibleAssetValue.FromRawValue(Asset.Share, value.RawValue);
+        }
```

Comment:
`ShareFromPower` can make confusion, IMO.
`Power` is always equivalent to `GovernanceToken`, while `Share` isn't.
`Share` is conditionally equivalent to `GovernanceToken`, and existence of `ShareFromPower` method can make people to think those are exchangeable, though it's code for tests.

Reply:
Hmm, is there any way to fix this? I have no idea right now, or is there a proper method name?

Reply:
First, I think you're right.